general_bounds.c

```diff
@@ -0,0 +1,45 @@
+#include "include/common.h"
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+/***
+ * This program is a fusion of ``bounds.c``
+ * and ``set_bounds.c``.
+ * CHERI limits the range of addresses that may be dereferenced
+ * and reading the value outside the range will always give
+ * an "In-address space security exception". However, if we try to
+ * extend the bounderies of a capable pointer by setting them explicitly,
+ * the program fails on "riscv64" but not on "morello".
+ ***/
+
+int main()
+{
+	int32_t array[12] = {0};
+	int32_t *typed_array = &array;
+	u_int32_t bounds = 48;
+
+	printf("Explicitly setting the bounds outside the range ");
+	if (strcmp(ARCH, "Morello") == 0)
+	{
+		printf("does not cause any failure on %s. It automatically clears them.\n", ARCH);
+	}
+	else
```

Comment:
I'm not sure this is quite correct: we only know this is true for cheri/risc-v I think?

Reply:
Right.